client/js/router/index.js

```diff
@@ -73,4 +76,14 @@ const router = new VueRouter({
   routes,
 });
 
+router.beforeEach(async (to, _from, next) => {
+  // Make sure we try and reestablish a player's session if one exists
+  // We do this before the route resolves to preempt the reauth/logout logic
+  await store.dispatch('requestStatus', {
+    router,
+    route: to,
+  });
+  next();
+});
```

Comment:
This is where the magic happens now. the `to` param is the current route, so we can use that to determine which logic to run. This also has the benefit of being within the Vue Router lifecycle, prior to any component level navigation guards so they can handle auth logic passed this point.